satellite/accounting/live/live_test.go

```diff
@@ -14,35 +14,188 @@ import (
+	assert.True(t, ok)
+
+	projectIDs, sum, err := populateCache(ctx, cache)
+	require.NoError(t, err)
+
+	// make sure all of the "projects" got all space updates and got right totals
+	for _, projID := range projectIDs {
+		spaceUsed, err := cache.GetProjectStorageUsage(ctx, projID)
+		require.NoError(t, err)
+		assert.Equalf(t, sum, spaceUsed, "projectID %v", projID)
+	}
+
+	err = cache.ResetTotals(ctx)
+	require.NoError(t, err)
+
+	for _, projID := range projectIDs {
+		spaceUsed, err := cache.GetProjectStorageUsage(ctx, projID)
+		require.NoError(t, err)
+		assert.EqualValues(t, 0, spaceUsed)
+	}
+}
+
+func TestRedisCacheConcurrency(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	address, cleanup, err := redisserver.Start()
+	require.NoError(t, err)
+	defer cleanup()
+
+	config := Config{
+		StorageBackend: "redis://" + address + "?db=0",
+	}
+	cache, err := NewCache(zaptest.NewLogger(t).Named("live-accounting"), config)
+	require.NoError(t, err)
+
+	projectID := testrand.UUID()
+
 	const (
-		valuesListSize  = 1000
-		valueMultiplier = 4096
-		numProjects     = 200
+		numConcurrent = 100
+		inlineAmount  = 10
+		remoteAmount  = 10
 	)
+	expectedSum := (inlineAmount * numConcurrent) + (remoteAmount * numConcurrent)
+
+	var group errgroup.Group
+	for i := 0; i < numConcurrent; i++ {
+		group.Go(func() error {
+			return cache.AddProjectStorageUsage(ctx, projectID, inlineAmount, remoteAmount)
+		})
+	}
+	require.NoError(t, group.Wait())
+
+	spaceUsed, err := cache.GetProjectStorageUsage(ctx, projectID)
+	require.NoError(t, err)
+
+	require.EqualValues(t, expectedSum, spaceUsed)
+}
+
+func TestPlainMemoryCacheConcurrency(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	config := Config{
+		StorageBackend: "plainmemory",
+	}
+	cache, err := NewCache(zaptest.NewLogger(t).Named("live-accounting"), config)
+	require.NoError(t, err)
+
+	projectID := testrand.UUID()
+
+	const (
+		numConcurrent = 100
+		inlineAmount  = 10
+		remoteAmount  = 10
+	)
+	expectedSum := (inlineAmount * numConcurrent) + (remoteAmount * numConcurrent)
+
+	var group errgroup.Group
+	for i := 0; i < numConcurrent; i++ {
+		group.Go(func() error {
+			return cache.AddProjectStorageUsage(ctx, projectID, inlineAmount, remoteAmount)
+		})
+	}
+	require.NoError(t, group.Wait())
+
+	spaceUsed, err := cache.GetProjectStorageUsage(ctx, projectID)
+	require.NoError(t, err)
+
+	require.EqualValues(t, expectedSum, spaceUsed)
+}
+
+func TestResetTotals(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
 	config := Config{
 		StorageBackend: "plainmemory:",
 	}
-	service, err := New(zaptest.NewLogger(t).Named("live-accounting"), config)
+	cache, err := NewCache(zaptest.NewLogger(t).Named("live-accounting"), config)
 	require.NoError(t, err)
 
 	// ensure we are using the expected underlying type
-	_, ok := service.(*plainMemoryLiveAccounting)
-	require.True(t, ok)
+	_, ok := cache.(*plainMemoryLiveAccounting)
+	assert.True(t, ok)
```

Comment:
Why not using [`IsType`](https://godoc.org/github.com/stretchr/testify/assert#IsType)?